src/community/security/oauth2-openid-connect/oauth2-openid-connect-core/src/main/java/org/geoserver/security/oauth2/OpenIdConnectAuthenticationFilter.java

```diff
@@ -265,4 +268,31 @@ private void enrichWithRoleCalculator(List<GeoServerRole> roles) throws IOExcept
         calc.addInheritedRoles(roles);
         calc.addMappedSystemRoles(roles);
     }
+
+    @Override
+    public void logout(
+            HttpServletRequest request,
+            HttpServletResponse response,
+            Authentication authentication) {
+
+        String idToken = null;
+        if (request.getAttribute(OpenIdConnectAuthenticationFilter.ID_TOKEN_VALUE) != null) {
+            idToken =
+                    (String) request.getAttribute(OpenIdConnectAuthenticationFilter.ID_TOKEN_VALUE);
+        } else {
+            OAuth2AccessToken token = restTemplate.getOAuth2ClientContext().getAccessToken();
+            if (token != null && token.getAdditionalInformation() != null) {
+                Object maybeIdToken = token.getAdditionalInformation().get("id_token");
+                if (maybeIdToken instanceof String) {
+                    idToken = (String) maybeIdToken;
+                }
+            }
+        }
```

Comment:
How comes there are two possible places where to look up this the idToken?

Reply:
I might be wrong but I guess probably because the request attribute has request scope, while the restTemplate is a bean with session scope.
When logging out from UI probably is likely that the fallback logic will be applied.

Reply:
As @taba90 said!